hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip551/AtomicBatchMiscSignatureTest.java

```diff
@@ -0,0 +1,196 @@
+import static com.hedera.services.bdd.suites.HapiSuite.DEFAULT_PAYER;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_HBAR;
+import static com.hedera.services.bdd.suites.HapiSuite.ONE_HUNDRED_HBARS;
+import static com.hedera.services.bdd.suites.hip869.NodeCreateTest.generateX509Certificates;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.INNER_TRANSACTION_FAILED;
+
+import com.hedera.services.bdd.junit.HapiTest;
+import com.hedera.services.bdd.junit.HapiTestLifecycle;
+import com.hedera.services.bdd.junit.support.TestLifecycle;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.Nested;
+
+@HapiTestLifecycle
+public class AtomicBatchMiscSignatureTest {
+
+    @BeforeAll
+    static void beforeAll(@NonNull final TestLifecycle testLifecycle) {
+        testLifecycle.overrideInClass(
+                Map.of("atomicBatch.isEnabled", "true", "atomicBatch.maxNumberOfTransactions", "50"));
+    }
+
+    @Nested
+    @DisplayName("Airdrop Tests")
+    class AirdropTests {
+
+        @HapiTest
+        @DisplayName("airdrop with missing sender's signature fails")
+        final Stream<DynamicTest> missingSenderSigFails() {
+            final var airdropOp =
+                    tokenAirdrop(moving(1, "FT").between("owner", "receiver")).batchKey("batchOperator");
+
+            return hapiTest(
+                    cryptoCreate("batchOperator"),
+                    cryptoCreate("owner"),
+                    cryptoCreate("receiver").maxAutomaticTokenAssociations(-1),
+                    tokenCreate("FT").treasury("owner"),
+                    // The owner is not signing the airdrop transaction, which should fail
+                    atomicBatch(airdropOp).payingWith("batchOperator").hasKnownStatus(INNER_TRANSACTION_FAILED));
+        }
+
+        @HapiTest
+        @DisplayName("claim airdrop with missing sender's signature fails")
+        final Stream<DynamicTest> claimSignedNotByTheReceiverFails() {
+            return hapiTest(
+                    cryptoCreate("batchOperator"),
+                    cryptoCreate("owner"),
+                    cryptoCreate("receiver").maxAutomaticTokenAssociations(0),
+                    cryptoCreate("fakeReceiver").maxAutomaticTokenAssociations(-1),
+                    tokenCreate("FT").treasury("owner"),
+                    tokenAirdrop(moving(1, "FT").between("owner", "receiver")).payingWith("owner"),
+                    // Verify the airdrop is pending
+                    getAccountBalance("receiver").hasTokenBalance("FT", 0L),
+                    atomicBatch(
+                                    // Payer is not the receiver
```

Comment:
Thanks for these little comments! It doesn't take long to find the code not signed by the receiver (as the title indicates), but having a comment like this still saves a few seconds of processing time :)